src/main/java/edu/upf/taln/dri/lib/model/util/GROBIDAuthorExtractor.java

```diff
@@ -0,0 +1,142 @@
+package edu.upf.taln.dri.lib.model.util;
```

Comment:
The if-block in `ObjectGenerator` was extracted into a method, and then moved into this new class.
The signature was changed to not pass in the `resultImpl`, but rather just return a `List` of the authors extracted.

The code inside is mostly identical. (We took some suggestions from IntelliJ, which removed some redundant predicates in the internal if-blocks)